services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/logging/CheckLoggingSpec.groovy

```diff
@@ -1,9 +1,74 @@
 package org.openkilda.functionaltests.spec.logging
 
+import groovy.util.logging.Slf4j
 import org.openkilda.functionaltests.BaseSpecification
+import org.openkilda.functionaltests.helpers.Wrappers
+import org.openkilda.testing.service.elastic.ElasticQueryBuilder
+import org.openkilda.testing.service.elastic.ElasticService
+import org.springframework.beans.factory.annotation.Autowired
 import spock.lang.Narrative
 
+@Slf4j
 @Narrative("This specification ensures that all logging facilities are up and running after Kilda deployment")
 class CheckLoggingSpec extends BaseSpecification {
 
+    @Autowired
+    ElasticService elasticService
+
+
+    def "Check Floodlight logging"() {
+        when: "Retrieve floodlight logs for last 5 minutes"
+        def tagName = "kilda-floodlight"
+        def result = elasticService.getLogs(new ElasticQueryBuilder().setTags(tagName).setTimeRange(300).build())
+        assert result?.hits?.total > 0
+
+        then: "There should be discovery messages"
+        result.hits.hits.any {hit -> hit.source.message.toLowerCase().contains("push discovery package via")}
+    }
+
+    def "Check Northbound, Storm and Topology Engine logging"() {
+        setup: "Request a non-existent flow"
+        def flowId = "nonexistentflowid" + System.currentTimeMillis()
+        try {
+            northbound.getFlow(flowId)
+        } catch (e) {
+        }
+        and: "Validate rules on a switch"
+        def switchId = topologyDefinition.switches[2].dpId
+        northbound.validateSwitchRules(switchId)
+        when: "Northbound, Storm and Topology Engine logs for 5 minutes are retrieved"
+        def nbTag = "kilda-northbound"
+        def stormTag = "storm-worker_log"
+        def tpTag = "kilda-tpe"
+
+        def nbResult = false
+        def stormResult = false
+        def tpResult = false
+        def timeout = 60
+
+        Wrappers.wait(timeout, 10) {
+            def nbLogs = elasticService.getLogs(new ElasticQueryBuilder().setTags(nbTag).setTimeRange(timeout * 2).
+                    setLevel("ERROR").build())
+            def stormLogs = elasticService.getLogs(new ElasticQueryBuilder().setTags(stormTag).
+                    setTimeRange(timeout * 2).setLevel("ERROR").build())
+            def tpLogs = elasticService.getLogs(new ElasticQueryBuilder().setTags(tpTag).setTimeRange(timeout * 2).
+                    setLevel("INFO").build())
+
+            if ([nbLogs, stormLogs, tpLogs].every {it?.hits?.total > 0})
+            {
+                nbResult = nbLogs.hits.hits.any { hit -> hit.source.message.contains(flowId) }
+                stormResult = stormLogs.hits.hits.any { hit -> hit.source.message.contains(flowId) }
+                tpResult = tpLogs.hits.hits.any {hit -> hit.source.message.contains(switchId.toString())}
+            }
+            [nbLogs, stormLogs].every {it.hits.hits.any { hit -> hit.source.message.contains(flowId) }}
```

Comment:
Sorry I don't understand why this operation is performed twice

Reply:
Moreover, the result of this operation is not saved and is literally ignored (this line does absolutely 100% nothing)

Reply:
The line has slipped during refactoring